crates/primitives/starknet/src/crypto/hash/pedersen.rs

```diff
@@ -20,10 +20,23 @@ impl HasherT for PedersenHasher {
     /// # Returns
     /// The hash of the data.
     fn hash_bytes(&self, data: &[u8]) -> Felt252Wrapper {
-        // For now we use the first 31 bytes of the data as the field element, to avoid any panics.
-        // TODO: have proper error handling and think about how to hash efficiently big chunks of data.
-        let field_element = FieldElement::from_byte_slice_be(&data[..31]).unwrap();
-        Felt252Wrapper(pedersen_hash(&FieldElement::ZERO, &field_element))
+        // Calculate the number of 31-byte chunks we'll need, rounding up.
+        // (1 byte is used padding to prevent the value of field from being greater than the field's
+        // modulus) TODO: It is need a way to truncate bytes to fit into values smaller than modular
+        // (for optimization)
+        const CHUNK_SIZE: usize = 31;
+        let mut hash_value = FieldElement::ZERO;
+
+        for chunk in data.chunks(CHUNK_SIZE) {
+            // It is safe to unwrap here because we know that the chunk size is 31 and the value can not
+            // overflow than the field's modulus value. In more detail, the FieldElement Maximum value is 2^251
+            // + 17 * 2^192. So the chunk (31 bytes is 248 bits) is smaller than the maximum value (== 2^248 - 1
+            // < 2^251 + 17 * 2^192). So it is safe to unwrap here.
+            let field_element = FieldElement::from_byte_slice_be(chunk).unwrap();
+            hash_value = pedersen_hash(&hash_value, &field_element);
+        }
+
+        Felt252Wrapper(hash_value)
```

Comment:
In my opinion, the `hash_bytes_in_chunks` function is unnecessary because `hash_bytes` handles bytes of various lengths and `CHUNK_SIZE` can be a maximum of 32 to handle efficiently. 

However, if you give `CHUNK_SIZE` as 32, the value can be 2^256 - 1, which can exceed the maximum value of the FieldElement and cause an error. Therefore, I set `CHUNK_SIZE` to 31, which is 2^248 - 1, which does not exceed the maximum value of the FieldElement and processes it with maximum efficiency. So, I think the `hash_bytes_in_chunks` function is not necessary to handle CHUNK_SIZE as a separate parameter.

Reply:
@apoorvsadana I don't think that is an issue here. The logic we are using for computing the hash_bytes is different in pedersen and poseidon. Trying to find common ground will result in an over-engineered solution

Reply:
Understoof @stompesi. Ya makes sense @d-roak, looks good then